website/cue/reference/remap/functions/get_enrichment_table_record.cue

```diff
@@ -0,0 +1,80 @@
+package metadata
+
+remap: functions: get_enrichment_table_record: {
+	category:    "Enrichment"
+	description: """
+		Searches an [enrichment table](\(urls.enrichment_tables_concept)) for a row that matches the
+		provided condition. A single row must be matched. If either no rows or more than one row is
+		found, an error is returned.
+
+		\(remap._enrichment_table_explainer)
+		"""
+
+	arguments: [
+		{
+			name:        "table"
+			description: "The [enrichment table](\(urls.enrichment_tables_concept)) to search."
+			required:    true
+			type: ["string"]
+		},
+		{
+			name: "condition"
+			description: """
+				The condition to search on. Since the condition is used at boot time to create
+				indices into the data, these conditions must be statically defined.
+				"""
+			required: true
+			type: ["object"]
+		},
+		{
+			name: "select"
+			description: """
+				A subset of fields from the enrichment table to return. If not specified,
+				all fields are returned.
+				"""
+			required: false
+			type: ["array"]
+		},
+		{
+			name:        "case_sensitive"
+			description: "Should text fields match case exactly."
+			required:    false
+			type: ["boolean"]
+			default: true
+		},
+	]
+	internal_failure_reasons: [
+		"The row isn't found",
+		"Multiple rows are found that match the condition",
+	]
+	return: types: ["object"]
+
+	examples: [
```

Comment:
Given there are two error states, and each state has its own error message, it might be worth providing an example on how to differentiate between the two errors (e.g. comparing `err` against both strings).

Reply:
I hesitate to push people to use string error matching since it is awkward and brittle (we make no compatibility guarantees about error strings). I think we should generally only provide examples of handling errors opaquely until we provide better utilities for differentiating errors.

Reply:
That makes sense to me.